charts/gateway-helm/templates/envoy-gateway-rbac.yaml

```diff
@@ -1,3 +1,123 @@
+{{ $allNamespaces := "*" }}
```

Comment:
pretty close, this looks good 
what I was referring to earlier is you will need to create a `ClusterRole` + `ClusterRoleBinding` for a `GatewayClass` by default / always since GatewayClass is always cluster scoped 
https://gateway-api.sigs.k8s.io/api-types/gatewayclass/

Reply:
do you see the same error if you replace `Role` with `ClusterRole` , but use `RoleBinding` ?

Reply:
yes, the error is same

Reply:
hmm any ideas @envoyproxy/gateway-reviewers & @envoyproxy/gateway-maintainers ?

Reply:
just thinking out loud, whether it has something to do with the scope of the [manager](https://github.com/envoyproxy/gateway/blob/7c6aa9ea849667d40d0c74c2acf37c8664068329/internal/provider/kubernetes/kubernetes.go#L51). not sure though. The `at the cluster scope` pointed me in that direction.

Reply:
it has been tested out using https://gateway.envoyproxy.io/v0.4.0/user/deployment-mode.html#multi-tenancy, and it works great!